eventhubsprocessor/partition_manager.py

```diff
@@ -0,0 +1,280 @@
+                            our_lease_count += 1
+                        else:
+                            leases_owned_by_others.append(possible_lease)
+
+                    elif possible_lease.owner == self.host.host_name:
+                        try:
+                            logging.info("Trying to renew lease %s %s", self.host.guid,
+                                         possible_lease.partition_id)
+                            if await lease_manager.renew_lease_async(possible_lease):
+                                our_lease_count += 1
+                            else:
+                                leases_owned_by_others.append(possible_lease)
+                        except Exception as err: #Update to LeaseLostException:
+                            logging.error("Lease lost exception %s %s %s", repr(err),
+                                          self.host.guid, possible_lease.partition_id)
+                            leases_owned_by_others.append(possible_lease)
+                    else:
+                        leases_owned_by_others.append(possible_lease)
+
+                except Exception as err:
+                    logging.error("Failure during getting/acquiring/renewing lease,\
+                                  skipping %s", repr(err))
+
+            # Grab more leases if available and needed for load balancing
+            leases_owned_by_others_count = len(leases_owned_by_others)
+            if leases_owned_by_others_count > 0:
+                steal_this_lease = self.which_lease_to_steal(leases_owned_by_others,
+                                                             our_lease_count)
+                if steal_this_lease:
+                    try:
+                        logging.info("Lease to steal %s", str(steal_this_lease.serializable()))
+                        if await lease_manager.acquire_lease_async(steal_this_lease):
+                            logging.info("Stole lease sucessfully %s %s", self.host.guid,
+                                         steal_this_lease.partition_id)
+                        else:
+                            logging.info("Failed to steal lease for partition %s %s",
+                                         self.host.guid, steal_this_lease.partition_id)
+                    except Exception as err:
+                        logging.error("Failed to steal lease %s", repr(err))
+
+            for partition_id in all_leases:
+                try:
+                    updated_lease = all_leases[partition_id]
+                    if updated_lease.owner == self.host.host_name:
+                        await self.check_and_add_pump_async(partition_id, updated_lease)
+                    else:
+                        await self.remove_pump_async(partition_id, "LeaseLost")
+                except Exception as err:
+                    logging.error("failed to update lease %s", err)
+
+                # time.sleep(lease_manager.lease_renew_interval)
```

Comment:
Need a delay here. maybe "await asyncio.sleep(renew_interval)"?

Reply:
I was wondering about that will put it in. What is the delay used for?

Reply:
With a delay the lease management runs in a tight loop which is not needed.

Reply:
That makes sense +1